### PR TITLE
[SYCL] Introduce SYCL_PI_SUPPRESS_ERROR_MESSAGE

### PR DESCRIPTION
**Pull request by jchlanda**
Prevent `check_error` from printing error message. This is handy in CI where we would like to match on messages generated by the underlying driver, over which we have no control, and `check_error` could garble that output.

Fixes: https://github.com/intel/llvm/issues/6416

### CODE REVIEW

**Comment by jchlanda** [comment]
/verify with: https://github.com/intel/llvm-test-suite/pull/1089

File: sycl/doc/EnvironmentVariables.md [hunk 1]
```diff
@@ -169,6 +169,7 @@ variables in production code.</span>
 | Environment variable | Values | Description |
 | -------------------- | ------ | ----------- |
 | `SYCL_PI_CUDA_ENABLE_IMAGE_SUPPORT` (experimental) | Any(\*) | Enable support of images. This option is experimental since the image support is not fully implemented. |
+| `SYCL_PI_SUPPRESS_ERROR_MESSAGE` (experimental) | Any(\*) | Suppress printing of error message, only used for CI in order not to interrupt errors generated by underlying toolchains. |
```
[the remark this answers is not in the document]
Comment (jchlanda) [reply sycl/doc/EnvironmentVariables.md]: Yeap, I think you are right, I'll move it to `Debugging variables for DPC++ Runtime` and add clarifications.

**Review by smaslov-intel — APPROVED** [review]

**Comment by bader** [comment]
> Fixes: #6416

I think we need to set this variable in the test to fix #6416. Right?

**Comment by jchlanda** [comment]
> > Fixes: #6416
> 
> I think we need to set this variable in the test to fix #6416. Right?

Yeap, it's been done here: https://github.com/intel/llvm-test-suite/pull/1089/files

**Pull request merged**